.buildkite/serve.rayci.yml

```diff
@@ -7,11 +7,18 @@ steps:
     depends_on: servebuild
     job_env: forge
 
-  - label: ":ray-serve: serve: serve tests"
+  - label: ":ray-serve: serve: unit tests"
+    instance_type: small
+    commands:
+      - bazel run //ci/ray_ci:test_in_docker -- //python/ray/serve/tests/unit:all serve
+    depends_on: servebuild
+    job_env: forge
+
+  - label: ":ray-serve: serve: integration tests"
     parallelism: 2
     instance_type: large
     commands:
-      - bazel run //ci/ray_ci:test_in_docker -- //python/ray/serve/... serve --except-tags post_wheel_build,gpu,xcommit
+      - bazel run //ci/ray_ci:test_in_docker -- //python/ray/serve/tests:all serve --except-tags post_wheel_build,gpu,xcommit
```

Comment:
Looking good! But I wonder if we want to merge all these into just one job:

- bazel run //ci/ray_ci:test_in_docker -- //python/ray/serve/tests:all //python/ray/serve/tests/unit:all //python/ray/tests/..  serve --except-tags post_wheel_build,gpu,xcommit,worker-container

The overhead of creating one buildkite job is about 10 minutes (including 5 minutes downloading a docker image + 5 minutes of building ray). Because of that overhead, having many jobs each running a small number of tests is less effective 

You see that the core tests and unit test jobs takes about 12 minutes, but only 2 minute of those are running tests

Reply:
I see, the main reason I wanted to separate them is to have a fast/high fidelity signal in CI by getting unit test results quickly (without having to click in & follow the build). I think there's some utility there, but will defer to you.

Reply:
I reverted this part

Reply:
Sorry I was out during the weekend. This looks great, thanks for being flexible!